psi4/driver/driver_nbody.py

```diff
@@ -348,7 +348,7 @@ def nbody_gufunc(func: Union[str, Callable], method_string: str, **kwargs):
                     try:
                         wfn.set_scalar_variable(str(var), value)
                         core.set_scalar_variable(str(var), value)
-                    except:
+                    except Exception:
```

Comment:
My guess based on what _happens_ in the except cause would be that `value` is not the double expected by `set_scalar_variable`, so the code assumes it's a NumPy array and casts to matrix. This may happen, for example, if the quantity is a gradient.

I'd need to check what error gets raised in this case.

Reply:
This should be a `TypeError`.